### ISSUE TITLE
device_setpoint UI shows proportional boost value instead of direct target in mixed rooms (direct AC + proportional TRV)

### ISSUE THREAD

**Issue by florianmulatz** [snazzybean/roommind#327]
### RoomMind version

1.7.3

### Home Assistant version

2026.5.4

## Description

In a room configured with a **direct-mode AC** and a **proportional-mode TRV**, the UI displays an incorrect "Device set to X°C" value. It shows the proportional boost setpoint (e.g. `18.0°C`) while the AC actually receives the comfort target temperature (e.g. `22.0°C`).

## Root Cause

In `coordinator.py`, `_all_direct` is computed across **all** devices in the room:

```python
_direct_eids = get_direct_setpoint_eids(_room_devices)       # only the AC
_devs_with_eid = [d for d in _room_devices if d.get("entity_id")]  # AC + TRV
_all_direct = bool(_devs_with_eid) and len(_direct_eids) == len(_devs_with_eid)
# mixed room → _all_direct = False
```

`_compute_device_setpoint` only shortcuts to `target_temp` when `all_direct=True`. With `all_direct=False` it falls through to the proportional cooling formula:

```python
# coordinator.py, _compute_device_setpoint, cooling branch:
boost = device_min_temp          # e.g. 18.0°C (Daikin min_temp)
sp = round(current_temp - power_fraction * (current_temp - boost), 1)
sp = max(boost, sp)
sp = min(target_temp, sp)
# → 25.6 - 1.0 × (25.6 - 18.0) = 18.0°C  ← displayed in UI
```

Meanwhile in `mpc_controller.py`, `async_apply` correctly routes the AC to the direct value:

```python
ha_cool_direct = celsius_to_ha_temp(self.hass, effective_target)  # 22.0°C
ha_t = ha_cool_direct if eid in self._direct_eids else ha_target
# direct AC → receives 22.0°C
```

The UI calculation does not account for the fact that TRVs are turned off during cooling (`set_hvac_mode: off`), making them thermally inactive — yet their presence in the device list causes `_all_direct = False` and forces the proportional formula.

## Suggested Fix

`_compute_device_setpoint` (or its call site in `_build_room_state`) should evaluate directness only for the **AC subset** when computing the cooling display value, not across all devices:

```python
_ac_eids = {d["entity_id"] for d in _room_devices if d.get("type") == "ac" and d.get("entity_id")}
_acs_all_direct = bool(_ac_eids) and _ac_eids <= _direct_eids
```

When `_acs_all_direct` is `True` and mode is `MODE_COOLING`, return `target_temp` directly instead of the proportional formula.

## Additional Context

- Affected rooms: any room with `setpoint_mode: direct` AC + `setpoint_mode: proportional` TRV (common configuration when using RoomMind with a Daikin/similar AC alongside a heating TRV)
- The actual device control is **correct** — only the UI display value is wrong
- Relatedly: issue #241 describes EKF corruption when a direct-mode device doesn't fire due to internal hysteresis; the root cause (TRV presence masking direct AC behavior) has conceptual overlap

### Steps to reproduce

1. Configure a room with:
   - AC: `setpoint_mode: direct`
   - TRV: `setpoint_mode: proportional`
2. Room temperature significantly above comfort target (e.g. room 25.6°C, comfort target 22°C, `power_fraction` near 1.0)
3. Observe the "Device set to X°C" value in the RoomMind UI

**Expected:** `22.0°C` (the value actually sent to the AC)
**Actual:** `18.0°C` (proportional boost formula result, driven down to `device_min_temp`)

### Relevant log output

```text

```

**Comment by snazzybean** [comment]
Fixed in v1.7.6 (commit cc6244a). `_build_room_state_dict()` now evaluates setpoint directness only over the devices that are actually commanded in the current mode (cooling → ACs only), so a proportional TRV in the room no longer forces the direct AC's display value through the proportional boost formula.

Duplicate of this: #385, closing that one too.
